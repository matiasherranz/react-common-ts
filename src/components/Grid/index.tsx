import React, { ReactNode, ReactElement } from 'react';
import classNames from 'classnames';

export type Props = {
  children?: ReactNode;
  className?: string;
  gutterSize?: 'small' | 'medium' | 'large';
  verticalGutter?: boolean;
  outerGutter?: boolean;
};

const Grid = ({
  children,
  className,
  gutterSize = 'large',
  verticalGutter,
  outerGutter,
}: Props): ReactElement => {
  const classes = classNames(
    {
      'vertically-spaced-grid': verticalGutter,
      [`${gutterSize}-gutter-grid`]: gutterSize,
      'no-outer-gutter': !outerGutter,
    },
    className,
    'grid container-fluid'
  );

  return <div className={classes}>{children}</div>;
};

export default Grid;
