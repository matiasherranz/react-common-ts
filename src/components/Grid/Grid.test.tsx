import React from 'react';
import { render } from '@testing-library/react';

import Grid from './index';

it('renders without crashing', () => {
  const { container } = render(<Grid />);
  expect(container).toMatchSnapshot();
});
