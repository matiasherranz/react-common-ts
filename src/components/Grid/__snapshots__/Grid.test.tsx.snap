// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="large-gutter-grid no-outer-gutter grid container-fluid"
  />
</div>
`;
